tests/python_package_test/test_dask.py

```diff
@@ -36,6 +44,102 @@ def listen_port():
 listen_port.port = 13000
 
 
+def r2_score(dy_true, dy_pred):
+    """Helper function taken from dask_ml.metrics: computes coefficient of determination."""
+    numerator = ((dy_true - dy_pred) ** 2).sum(axis=0)
+    denominator = ((dy_true - dy_pred.mean(axis=0)) ** 2).sum(axis=0)
+    return (1 - numerator / denominator).compute()
```

Comment:
why was it necessary to replace the dask-ml one with this function?

If that's just to cut out our testing dependency on `dask-ml` (which would be much appreciated!), could you make that a separate PR with this change plus removing `dask-ml` from here:

* https://github.com/microsoft/LightGBM/blob/f2695daba2506d047831e39aeb521e94adec5e8d/.ci/test.sh#L71

I think that PR would be small and non-controversial, so we could review and merge it quickly.

Reply:
Cool, yeah I'll follow up 3708 with the removal of dask-ml as a test dependency. See reply to comment re: accuracy_score (above).

Reply:
Addressed in d6e5209

Reply:
hey @ffineis is it cool if I pick up the "remove testing dependency on `dask-ml`" thing?

If you want to make the contribution I'll totally respect that, but if it's just extra work you'd rather not do, I'm happy to make a PR right now for it. Wanna be respectful of your time.